File: app/src/main/java/com/example/bindechexclock/ui/dashboard/DashboardViewModel.java
package com.example.bindechexclock.ui.dashboard;

import androidx.lifecycle.LiveData; // Changed
import androidx.lifecycle.MutableLiveData; // Changed
import androidx.lifecycle.ViewModel; // Changed

public class DashboardViewModel extends ViewModel {

    private MutableLiveData<String> mText;

    public DashboardViewModel() {
        mText = new MutableLiveData<>();
        // mText.setValue("This is dashboard fragment"); // Default text removed
    }

    public LiveData<String> getText() {
        return mText;
    }

    public void setmText(String s) {
        mText.setValue(s);
    }
}

File: app/src/main/java/com/example/bindechexclock/ui/dashboard/DashboardViewModel.kt
package com.example.bindechexclock.ui.dashboard

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class DashboardViewModel : ViewModel() {

    private val _text = MutableLiveData<String>()
    val text: LiveData<String> get() = _text

    init {
        // If there was a default value like "This is dashboard fragment"
        // it would be set here, e.g., _text.value = "This is dashboard fragment"
        // For now, it's initialized by DashboardFragment.
    }

    fun updateText(newText: String) {
        _text.value = newText
    }
}
